File: test_functions.py
import pytest

from functions import fibonacci_generator, vowel_counter, reverse_sentence, is_palindrome, fahrenheit_to_celsius

def test_fibonacci_generator():
    assert fibonacci_generator(3) == 1
    assert fibonacci_generator(5) == 3
    assert fibonacci_generator(10) == 34
    assert fibonacci_generator(20) == 4181

def test_vowel_counter():
    assert vowel_counter("hello") == 2
    assert vowel_counter("hello world") == 3
    assert vowel_counter("this is a sentence with vowels") == 8

def test_reverse_sentence():
    assert reverse_sentence("hello world") == "world hello"
    assert reverse_sentence("this is a sentence") == "sentence a is this"
    assert reverse_sentence("this is a sentence with words") == "words with sentence a is this"

def test_is_palindrome():
    assert is_palindrome("racecar") == True
    assert is_palindrome("hello") == False
    assert is_palindrome("hannah") == True
    assert is_palindrome("race car") == True
    assert is_palindrome("a h i ha") == True

def test_fahrenheit_to_celsius():
    assert fahrenheit_to_celsius(32) == 0
    assert fahrenheit_to_celsius(212) == 100
    assert fahrenheit_to_celsius(98.6) == 37
    assert fahrenheit_to_celsius(0) == -17.77777777777778

File: test_functions.R
install.packages("testthat", repos = "http://cran.us.r-project.org")

library(testthat)

source("functions.R")

test_that("fahr_to_celsius function", {
  expect_equal(fahr_to_celsius(32), 0)
  expect_equal(fahr_to_celsius(212), 100)
  expect_equal(fahr_to_celsius(98.6), 37)
})

test_that("fibonacci_generator function", {
  expect_equal(fibonacci_generator(3), 1)
  expect_equal(fibonacci_generator(5), 3)
  expect_equal(fibonacci_generator(10), 34)
  expect_equal(fibonacci_generator(20), 4181)
})

test_that("vowel_counter function", {
  expect_equal(vowel_counter("hello"), 2)
  expect_equal(vowel_counter("hello world"), 3)
  expect_equal(vowel_counter("this is a sentence with vowels"), 8)
})

test_that("reverse_sentence function", {
  expect_equal(reverse_sentence("hello world"), "world hello")
  expect_equal(reverse_sentence("this is a sentence"), "sentence a is this")
  expect_equal(reverse_sentence("this is a sentence with words"), "words with sentence a is this")
})

test_that("is_palindrome function", {
  expect_equal(is_palindrome("racecar"), TRUE)
  expect_equal(is_palindrome("hello"), FALSE)
  expect_equal(is_palindrome("hannah"), TRUE)
  expect_equal(is_palindrome("race car"), TRUE)
  expect_equal(is_palindrome("a h i ha"), TRUE)
})


